searchcore/src/vespa/searchcore/proton/server/proton.cpp

```diff
@@ -95,11 +95,16 @@ deriveCompactionCompressionThreads(const ProtonConfig &proton,
 
     // We need at least 1 guaranteed free worker in order to ensure progress so #documentsdbs + 1 should suffice,
     // but we will not be cheap and give it one extra.
-    return std::max(scaledCores, proton.documentdb.size() + 1 + 1);
+    return std::max(scaledCores, proton.documentdb.size() + proton.flush.maxconcurrent + 1);
```

Comment:
While this will likely avoid deadlock in most cases, there are still issues if a document type is added.

Reply:
Correct, but that will be solved later on with a proper fix not relying on a minimum number of thread in the executor.